contracts/StablecoinConverter.sol

```diff
@@ -112,8 +115,14 @@ contract StablecoinConverter is EpochTokenLocker {
     }
 
     mapping (uint16 => uint128) public currentPrices;
+    mapping (uint32 => AuctionData) public auctionData;
```

Comment:
auctionData[batchIndex].feeCollected is needed to judge new solutions. I added the msg.sender, as it is important later for the withdraw. Yeah, I can take the bestSolutionSubmitter out, as it is only needed later for the withdraw of the reward

Reply:
Makes sense that we need history for later reward withdrawal (we probably don't need history on the fees though).
Would it make sense to pay the reward as soon as the first solution in the next batch is submitted?

Reply:
Actually speaking about rewards, isn't the reward that the solver credits themselves the fee surplus? This would automatically get reverted by a better solution so there shouldn't be a need to later claim rewards.

That being said, the part about crediting the difference is not yet implemented, but in my opinion it should be part of "apply solution".

Reply:
And we store it with history, as it is needed for the GNO withdrawals later from the solution submitter. Once the withdraw is done, we can set the history value to 0

Reply:
Why don't we credit the GNO to the msg.sender's balance just like any other trade execution?